src/openfermion/utils/_low_rank_test.py

```diff
@@ -0,0 +1,181 @@
+
+        # Convert one-body.
+        for p, q in itertools.product(range(n_qubits), repeat=2):
+            term = ((p, 1), (q, 0))
+            coefficient = fo_one_body_coefficients[p, q]
+            output_operator += FermionOperator(term, coefficient)
+
+        # Convert two-body.
+        for p, q, r, s in itertools.product(range(n_qubits), repeat=4):
+            term = ((p, 1), (q, 0), (r, 1), (s, 0))
+            coefficient = fo_chemist_tensor[p, q, r, s]
+            output_operator += FermionOperator(term, coefficient)
+
+        # Check that difference is small.
+        difference = normal_ordered(random_fermion - output_operator)
+        self.assertAlmostEqual(0., difference.induced_norm())
+
+    def test_exception(self):
+
+        # Initialize a bad FermionOperator.
+        n_qubits = 4
+        random_interaction = random_interaction_operator(n_qubits)
+        random_fermion = get_fermion_operator(random_interaction)
+        bad_term = ((1, 1), (2, 1))
+        random_fermion += FermionOperator(bad_term)
+
+        # Check for exception.
+        with self.assertRaises(TypeError):
+            fo_constant, fo_one_body_coefficients, fo_chemist_tensor = \
+                get_chemist_two_body_coefficients(random_fermion)
+
+
+class LowRankTest(unittest.TestCase):
+
+    def test_operator_consistency(self):
+
+        # Initialize a random two-body FermionOperator.
+        n_qubits = 4
+        random_operator = get_fermion_operator(
+            random_interaction_operator(n_qubits))
+
+        # Convert to chemist tensor.
+        constant, one_body_coefficients, chemist_tensor = \
+            get_chemist_two_body_coefficients(random_operator)
+
+        # Build back operator constant and one-body components.
+        decomposed_operator = FermionOperator((), constant)
+        for p, q in itertools.product(range(n_qubits), repeat=2):
+            term = ((p, 1), (q, 0))
+            coefficient = one_body_coefficients[p, q]
+            decomposed_operator += FermionOperator(term, coefficient)
+
+        # Perform decomposition.
+        eigenvalues, one_body_squares = low_rank_two_body_decomposition(
+            chemist_tensor)
+
+        # Build back two-body component.
+        for l in range(n_qubits ** 2):
+            one_body_operator = FermionOperator()
+            for p, q in itertools.product(range(n_qubits), repeat=2):
+                term = ((p, 1), (q, 0))
+                coefficient = one_body_squares[l, p, q]
+                one_body_operator += FermionOperator(term, coefficient)
+            decomposed_operator += eigenvalues[l] * (one_body_operator ** 2)
+
+        # Test for consistency.
+        difference = normal_ordered(decomposed_operator - random_operator)
+        self.assertAlmostEqual(0., difference.induced_norm())
+
+    def test_rank_reduction(self):
+
+        # Initialize H2.
+        geometry = [('H', (0., 0., 0.)), ('H', (0., 0., 0.7414))]
+        basis = 'sto-3g'
```

Comment:
we'd need to add that file to OpenFermion and also I think it would slow down testing unfortunately. Perhaps we can see real rank reduction once the UCC functionality is added?